pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -47,6 +47,8 @@ class NativeAssetsBuildRunner {
     required BuildModeImpl buildMode,
     CCompilerConfigImpl? cCompilerConfig,
     IOSSdkImpl? targetIOSSdk,
+    int? targetIOSVersion,
```

Comment:
While it's also nice to have a "flat" config of many top-level keys, they are becoming so many that it would be nice to combine them into objects if easily possible. This might make most of the `?`s and `StateError`s go away.

Reply:
Yeah, we probably should do another iteration on the API as it grows. Leaving this for another PR.